datafusion/sqllogictest/test_files/array.slt

```diff
@@ -557,10 +557,16 @@ select column1[0:5], column2[0:3], column3[0:9] from arrays;
 ## make_array (aliases: `make_list`)
 
 # make_array scalar function #1
-query ???
-select make_array(1, 2, 3), make_array(1.0, 2.0, 3.0), make_array('h', 'e', 'l', 'l', 'o');
-----
-[1, 2, 3] [1.0, 2.0, 3.0] [h, e, l, l, o]
+query ??????
```

Comment:
Can you also add some tests for other uses of this logic (not just in make_array) such as comparisons

I notice that postgres doesn't handle boolean coercion 

```
postgres=# select true = 1;
ERROR:  operator does not exist: boolean = integer
LINE 1: select true = 1;
                    ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
```

However, after this PR datafusion does:

```sql
DataFusion CLI v33.0.0
❯ select true = 1;
+--------------------------+
| Boolean(true) = Int64(1) |
+--------------------------+
| true                     |
+--------------------------+
1 row in set. Query took 0.006 seconds.
```

Reply:
Duckdb support boolean coercion, we can consider follow it. If it does not break the overall design, I think we can support it too.